src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -47,18 +47,31 @@ public function copy($originFile, $targetFile, $override = false)
         }
 
         if ($doCopy) {
-            // https://bugs.php.net/bug.php?id=64634
-            $source = fopen($originFile, 'r');
-            // Stream context created to allow files overwrite when using FTP stream wrapper - disabled by default
-            $target = fopen($targetFile, 'w', null, stream_context_create(array('ftp' => array('overwrite' => true))));
-            stream_copy_to_stream($source, $target);
-            fclose($source);
-            fclose($target);
-            unset($source, $target);
-
-            if (!is_file($targetFile)) {
-                throw new IOException(sprintf('Failed to copy %s to %s', $originFile, $targetFile));
-            }
+            $this->doCopy($originFile, $targetFile);
+        }
+    }
+
+    /**
+     * Makes of copy from the origin file to the target file.
+     *
+     * @param string $originFile The original filename
+     * @param string $targetFile The target filename
+     *
+     * @throws IOException When copy fails
+     */
+    protected function doCopy($originFile, $targetFile)
```

Comment:
Unfortunately, this would make the `doCopy` function unmockable. Does it really must be private? If so, another possible solution would be moving the decision itself to a separate private function, then test that function directly with different parameters using `Reflection` to make the private function accessible temporarily (just for the test).